Strategy11/formidable-forms
# Reduce server load and page load times

- Set a 2 minute transient to avoid duplicate requests
- Increase the time between checking for new templates
- Use site options for multisite update checks so every site in the network is not checking if they use the same license
- Don't check if the site url changed and needs to be activated if the last try was not successful
- Don't cache a rate limited response for the normal time. Allow the next check sooner.
- Use utc times for api calls instead of adjusting for utc
- Fix expired banner and php errors on form templates page when site is rate limited

Part of https://github.com/Strategy11/formidable-pro/issues/4813 (covers checked boxes in linked issue)

How to test:
- Force a rate limit response. This will add a line to the error log each time an API request to ff.com is triggered. When rate limited, the site will check every 5 minutes.
```
add_filter(
	'pre_http_request',
	function( $pre, $parsed_args, $url ) {
		if ( strpos( $url, 'formidableforms.com' ) === false ) {
			return $pre;
		}
		error_log( 'http_request: ' . $url );

		return array(
			'response' => array( 'code' => 429, 'message' => 'Limited' ),
			'body'     => '{"error":{"type":"basic","code":"valid","expires":"","message":"Your site has been rate limited for 1 minute. Your forms will continue to function properly. If this message persists, please click the link to Check for a recent purchase on the Settings page","license":"xxx"},"expires":""}',
		);
	},
	10,
	3
);
```
- Click "check for recent purchases" and watch the log with this snippet to see how often api calls are being triggered:
```
add_filter(
	'pre_http_request',
	function( $pre, $parsed_args, $url ) {
		if ( strpos( $url, 'formidableforms.com' ) === false ) {
			return $pre;
		}
		error_log( 'http_request: ' . $url );

		return $pre;
	},
	10,
	3
);
```
